test/telemetry_metrics_appsignal_test.exs

```diff
@@ -145,6 +145,54 @@ defmodule TelemetryMetricsAppsignalTest do
     assert_receive {^ref, :called}
   end
 
+  test "reporting calculated metrics with unary functions" do
+    # `Telemetry.Metrics.Summary` maps to AppSignal's measurement metric
+    metric =
+      summary("db.query.foo",
+        measurement: fn measurement ->
+          measurement.duration * 10
+        end
+      )
+
+    start_reporter(metrics: [metric])
+
+    parent = self()
+    ref = make_ref()
+
+    expect(AppsignalMock, :add_distribution_value, fn
+      "db.query.foo", 990, %{} ->
+        send(parent, {ref, :called})
+        :ok
+    end)
+
+    :telemetry.execute([:db, :query], %{duration: 99}, %{})
+    assert_receive {^ref, :called}
+  end
+
+  test "reporting calculated metrics with binary functions" do
+    # `Telemetry.Metrics.Summary` maps to AppSignal's measurement metric
+    metric =
+      summary("db.query.foo",
+        measurement: fn _measurement, metadata ->
+          metadata.some_metadata_value + 1
```

Comment:
Let's maybe use both measurements and metadata to test if everything is passed correctly. We could use a `multiplier` to multiply the duration like in the test case above:

```suggestion
        measurement: fn measurements, metadata ->
          measurements.duration * metadata.multiplier
```